__tests__/app.test.js

```diff
@@ -240,6 +240,47 @@ describe("POST /api/articles/:article_id/comments", () => {
         expect(msg).toBe("3432432 does not exist in articles");
       });
   });
+  test("POST:404 responds with an appropriate status and error message when provided a non-existing article id", () => {
+    const newComment = {
+      author: "butter_bridge",
+      body: "a new comment",
+    };
+    return request(app)
+      .post("/api/articles/3432432/comments")
+      .send(newComment)
+      .expect(404)
+      .then((response) => {
+        const msg = response.body.msg;
+        expect(msg).toBe("3432432 does not exist in articles");
+      });
+  });
+  test("POST:400 responds with an appropriate error message when given an invalid id", () => {
+    const newComment = {
+      author: "butter_bridge",
+      body: "a new comment",
+    };
+    return request(app)
+      .post("/api/articles/not-an-id/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad Request");
+      });
+  });
+  test("POST:404 responds with an appropriate status and error message when provided a non-existing username", () => {
+    const newComment = {
+      author: "fdsfdsf",
+      body: "a new comment",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(newComment)
+      .expect(404)
+      .then((response) => {
+        const msg = response.body.msg;
+        expect(msg).toBe("fdsfdsf does not exist in users");
+      });
+  });
```

Comment:
Great tests, however, we also need to look to add
Status 201, ignores unnecessary properties
Status 400, missing required field(s), e.g. no username or body properties